src/frontend/screens/Audio/ExistingRecording.tsx

```diff
@@ -0,0 +1,191 @@
+import React, {useEffect, useState} from 'react';
+import {Pressable, View, StyleSheet} from 'react-native';
+import {
+  HeaderBackButton,
+  HeaderBackButtonProps,
+} from '@react-navigation/elements';
+import {useNavigation} from '@react-navigation/native';
+import {useIntl, defineMessages} from 'react-intl';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
+import {Playback} from '../../sharedComponents/Playback';
+import {
+  BottomSheetModal,
+  BottomSheetModalContent,
+  useBottomSheetModal,
+} from '../../sharedComponents/BottomSheetModal';
+import {CloseIcon, DeleteIcon} from '../../sharedComponents/icons';
+import {WHITE, BLACK} from '../../lib/styles';
+import ErrorIcon from '../../images/Error.svg';
+import Share from 'react-native-share';
+import * as FileSystem from 'expo-file-system';
+import {UIActivityIndicator} from 'react-native-indicators';
+
+const m = defineMessages({
+  deleteBottomSheetTitle: {
+    id: 'screens.AudioScreen.ExistingRecording.deleteBottomSheetTitle',
+    defaultMessage: 'Delete?',
+  },
+  deleteBottomSheetDescription: {
+    id: 'screens.AudioScreen.ExistingRecording.deleteBottomSheetDescription',
+    defaultMessage:
+      'Your Audio Recording will be permanently deleted. This cannot be undone.',
+  },
+  deleteBottomSheetPrimaryButtonText: {
+    id: 'screens.AudioScreen.ExistingRecording.deleteBottomSheetPrimaryButtonText',
+    defaultMessage: 'Delete',
+  },
+  deleteBottomSheetSecondaryButtonText: {
+    id: 'screens.AudioScreen.ExistingRecording.deleteBottomSheetSecondaryButtonText',
+    defaultMessage: 'Cancel',
+  },
+});
+
+interface ExistingRecordingProps {
+  uri: string;
+  onDelete: () => void;
+  isEditing: boolean;
+}
+
+export const ExistingRecording: React.FC<ExistingRecordingProps> = ({
+  uri,
+  onDelete,
+  isEditing,
+}) => {
+  const {formatMessage: t} = useIntl();
+  const navigation = useNavigation();
+  const {sheetRef, isOpen, openSheet, closeSheet} = useBottomSheetModal({
+    openOnMount: false,
+  });
+
+  const [localUri, setLocalUri] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    let isCancelled = false;
+    const tempFileName = `audio_${Date.now()}.m4a`;
+    const localFilePath = `${FileSystem.cacheDirectory}${tempFileName}`;
+    setLoading(true);
+
+    const downloadAudio = async () => {
+      try {
+        const downloadResult = await FileSystem.downloadAsync(
+          uri,
+          localFilePath,
+        );
+        if (!isCancelled) {
+          setLocalUri(downloadResult.uri);
+        }
+      } catch (err) {
+        console.error('Error downloading audio file:', err);
+      } finally {
+        if (!isCancelled) {
+          setLoading(false);
+        }
+      }
+    };
+
+    downloadAudio();
+
+    return () => {
+      isCancelled = true;
+      if (localUri) {
+        FileSystem.deleteAsync(localUri, {idempotent: true}).catch(() => {});
```

Comment:
I understand that the mentality is to have the uri available as soon as they press share, and in order to do that we want to start loading it into memory as soon as this screen opens. But this will cause it to load into memory twice. On first render, local uri is not available, so `useAudioPlayback` will take the uri, and load it into memory with `Audio.Sound.createAsync()`. Then the useeffect you have written here will also start to load into memory. then when that is loaded into memory, a uri will be created, which will replace the original uri in `useAudioPlayback`. BUT useAudioPlayback will still call `Audio.Sound.createAsync()`on this locauri and start loading into memory again.

I think the original approach of only downloading it onClick of the share button avoids unnecessarily having it twice in memory (and yes it will still be in memory twice, but only if the user chooses to share it, not every single time)

Reply:
So I think I have fixed the issue and the crash. Sorry about that. I made the change because the Share was not working after the first Share... so I had to cache the uri (localURI -- now just used for sharing). I did it wrong in a rush at the end of the day. So,  I made a change to cache the uri when someone presses Share so that a user can share multiple times and it won't load twice or crash the playback! I am only deleting it on unmount ( I think I did that right but could use your eyes on it).